### ISSUE TITLE
sct_run_batch crashes on sub-mountSinai05

### ISSUE THREAD

**Issue by valosekj** [spine-generic/spine-generic#88]
When run analysis of latest snapshot of multi-subject dataset using:

`sct_run_batch -jobs -1 -path-data ds002902-download/ -path-output ds002902-download_results/ /usr/local/lib/spine-generic/processing/process_data.sh`

Analysis failed on error related to sub-mountSinai05 subject:

```
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/local/lib/sct_v4.3/python/envs/venv_sct/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/sct_v4.3/python/envs/venv_sct/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/usr/local/lib/sct_v4.3/scripts/sct_run_batch.py", line 140, in run_single
    assert res.returncode == 0, 'Processing of subject {} failed'.format(subject)
AssertionError: Processing of subject sub-mountSinai05 failed
"""
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/sct_v4.3/scripts/sct_run_batch.py", line 152, in <module>
    p.map(run_single, subject_dirs)
  File "/usr/local/lib/sct_v4.3/python/envs/venv_sct/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/local/lib/sct_v4.3/python/envs/venv_sct/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
AssertionError: Processing of subject sub-mountSinai05 failed
```
At ds002902-download_results/log directory I can see that some errors happened not only in sub-mountSinai05 subject, but also for another subjects, but only error in sub-mountSinai05 causes crash of whole sct_run_batch script.

sub-mountSinai05 subject analysis failed during DWI analysis:

```
-------------------------------------------------------------------------------
  Estimating motion on b=0 images...
-------------------------------------------------------------------------------

Input parameters:
  Input file ............ b0.nii
  Reference file ........ data_T0000.nii.gz
  Polynomial degree ..... 2
  Smoothing kernel ...... 1
  Gradient step ......... 1
  Metric ................ MI
  Sampling .............. None
  Todo .................. estimate_and_apply
  Mask  ................. 
  Output mat folder ..... mat_b0groups

Data dimensions:
  33 x 34 x 15 x 5

Copy file_target to a temporary file...

Register. Loop across Z (note: there is only one Z if orientation is axial)
Traceback (most recent call last):
  File "/usr/local/lib/sct_v4.3/scripts/sct_dmri_moco.py", line 155, in <module>
    main()
  File "/usr/local/lib/sct_v4.3/scripts/sct_dmri_moco.py", line 150, in main
    moco_wrapper(param)
  File "/usr/local/lib/sct_v4.3/spinalcordtoolbox/moco.py", line 318, in moco_wrapper
    file_mat_b0, _ = moco(param_moco)
  File "/usr/local/lib/sct_v4.3/spinalcordtoolbox/moco.py", line 607, in moco
    file_data_splitZ_splitT_moco[it], im_mask=input_mask)
  File "/usr/local/lib/sct_v4.3/spinalcordtoolbox/moco.py", line 746, in register
    status, output = sct.run(cmd, verbose=1 if param.verbose == 2 else 0, env=env, **kw)
  File "/usr/local/lib/sct_v4.3/scripts/sct_utils.py", line 310, in run
    raise RunError(output)
sct_utils.RunError: Exception caught:
itk::ExceptionObject (0x1f617b0)
Location: "unknown"
File: /__w/ANTs/ANTs/antsbin/staging/include/ITK-5.1/itkMattesMutualInformationImageToImageMetricv4.hxx
Line: 312
Description: itk::ERROR: MattesMutualInformationImageToImageMetricv4(0x1f563b0): Joint PDF summed to zero
```

**Comment by valosekj** [comment]
UPDATE:
dwi analysis on sub-mountSinai05 failed during `sct_dmri_moco`. I tried run `sct_dmri_moco` manually from SCT v4.2.2 and it passed successfully. So problem is somewhere in SCT v4.3.

**Comment by jcohenadad** [comment]
related to https://github.com/neuropoly/spinalcordtoolbox/issues/2725

**Comment by kousu** [comment]
> Description: itk::ERROR: MattesMutualInformationImageToImageMetricv4(0x1f563b0): Joint PDF summed to zero

This is due to https://github.com/neuropoly/spinalcordtoolbox/pull/2642. I saw this during testing and developing that but it was intermittent and I let it slide. It didn't crash our CI test cases so I figured it was okay. I'm sorry. I don't understand what the cause is.

Julien suggested the issue is one of the preprocessing detector steps is slightly less accurate now and that causes no overlap during the registration (i.e. image realignment) step.
